src/ekf_node.cpp

```diff
@@ -45,31 +42,12 @@
 int main(int argc, char ** argv)
 {
   rclcpp::init(argc, argv);
-  auto node = rclcpp::Node::make_shared("se_node");
-
-  std::string filter_type = node->declare_parameter("filter_type", "ekf");
-  std::transform(filter_type.begin(), filter_type.end(), filter_type.begin(),
-    ::tolower);
-
-  robot_localization::FilterBase::UniquePtr filter;
-
-  if (filter_type == "ukf") {
-    double alpha = node->declare_parameter("alpha", 0.001);
-    double kappa = node->declare_parameter("kappa", 0.0);
-    double beta = node->declare_parameter("beta", 2.0);
-
-    filter = std::make_unique<robot_localization::Ukf>(alpha, kappa, beta);
-  } else {
-    if (filter_type != "ekf") {
-      std::cerr << "Unsupported filter type of " << filter_type <<
-        " specified. Defaulting to ekf.\n";
-    }
-
-    filter = std::make_unique<robot_localization::Ekf>();
-  }
-
-  robot_localization::RosFilter ros_filter(node, filter);
-  ros_filter.run();
-
+  rclcpp::NodeOptions options;
+  options.arguments({"ekf_filter_node"});
+  std::shared_ptr<robot_localization::RosEkf> filter =
```

Comment:
Could this be an `auto` without loosing too much readability? Maybe I'm missing something on why you would like it to be an explicit type here? (again this is nitpicking so feel free to ignore)

Reply:
I hate auto, and its really not used in the styling of RL. I know there is some in there now, but I don't think that's something to be aimed for.